app/models/donation.rb

```diff
@@ -20,6 +21,22 @@ def self.current
     joins(:scheduled_pickup).merge(ScheduledPickup.current)
   end
 
+  def self.pending
+    where(confirmed_at: nil, declined_at: nil)
+  end
+
+  def self.scheduled_for_pick_up_within(hours:)
+    joins(:scheduled_pickup).
+      where(
+        "scheduled_pickups.start_at BETWEEN now() AND :cutoff",
```

Comment:
I often capitalize PostgreSQL keywords and function names to make them stand out vs. column attributes, so I would say `BETWEEN NOW() AND :cutoff` here for readability.